class-wxr-import-ui.php

```diff
@@ -0,0 +1,635 @@
+	protected function display_intro_step() {
+		require __DIR__ . '/templates/intro.php';
+	}
+
+	protected function render_upload_form() {
+		/**
+		 * Filter the maximum allowed upload size for import files.
+		 *
+		 * @since 2.3.0
+		 *
+		 * @see wp_max_upload_size()
+		 *
+		 * @param int $max_upload_size Allowed upload size. Default 1 MB.
+		 */
+		$max_upload_size = apply_filters( 'import_upload_size_limit', wp_max_upload_size() );
+		$upload_dir = wp_upload_dir();
+		if ( ! empty( $upload_dir['error'] ) ) {
+			$error = '<div class="error inline"><p>';
+			$error .= esc_html__(
+				'Before you can upload your import file, you will need to fix the following error:',
+				'wordpress-importer'
+			);
+			$error .= sprintf( '<p><strong>%s</strong></p></div>', $upload_dir['error'] );
+			echo $error;
+			return;
+		}
+
+		// Load the template
+		remove_action( 'post-plupload-upload-ui', 'media_upload_flash_bypass' );
+		require __DIR__ . '/templates/upload.php';
+		add_action( 'post-plupload-upload-ui', 'media_upload_flash_bypass' );
+
+		// Queue the JS needed for the page
+		$url = plugins_url( 'assets/intro.js', __FILE__ );
+		$deps = array(
+			'wp-backbone',
+			'wp-plupload',
+		);
+		wp_enqueue_script( 'import-upload', $url, $deps, false, true );
+
+		// Set uploader settings
+		wp_plupload_default_settings();
+		$settings = array(
+			'next_url' => wp_nonce_url( $this->get_url( 1 ), 'import-upload' ) . '&id={id}',
+			'plupload' => array(
+				'filter' => array(
+					'max_file_size' => $max_upload_size . 'b',
+				),
+
+				'file_data_name' => 'import',
+				'multipart_params' => array(
+					'action'   => 'wxr-import-upload',
+					'_wpnonce' => wp_create_nonce( 'wxr-import-upload' ),
+				),
+			),
+		);
+		wp_localize_script( 'import-upload', 'importUploadSettings', $settings );
+	}
+
+	/**
+	 * Display the author picker (or upload errors).
+	 */
+	protected function display_author_step() {
+		if ( isset( $_REQUEST['id'] ) ) {
+			$err = $this->handle_select( $_REQUEST['id'] );
+		} else {
+			$err = $this->handle_upload();
+		}
+		if ( is_wp_error( $err ) ) {
+			$this->display_error( $err );
+			return;
+		}
+
+		$data = $this->get_data_for_attachment( $this->id );
+		if ( is_wp_error( $data ) ) {
+			$this->display_error( $data );
+			return;
+		}
+
+		require __DIR__ . '/templates/select-options.php';
+	}
+
+	/**
+	 * Handles the WXR upload and initial parsing of the file to prepare for
+	 * displaying author import options
+	 *
+	 * @return bool|WP_Error True on success, error object otherwise.
+	 */
+	protected function handle_upload() {
+		$file = wp_import_handle_upload();
+
+		if ( isset( $file['error'] ) ) {
+			return new WP_Error( 'wxr_importer.upload.error', esc_html( $file['error'] ), $file );
+		} elseif ( ! file_exists( $file['file'] ) ) {
+			$message = sprintf(
+				__( 'The export file could not be found at <code>%s</code>. It is likely that this was caused by a permissions problem.', 'wordpress-importer' ),
```

Comment:
You could move the `<code>` tags to the line below and concat them around the filename, then you can use `esc_html__()` here instead